src/core_plugins/kibana/public/home/components/recently_accessed.js

```diff
@@ -0,0 +1,188 @@
+import './recently_accessed.less';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiPanel,
+  EuiLink,
+  EuiText,
+  EuiTextColor,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiIcon,
+} from '@elastic/eui';
+
+export const NUM_LONG_LINKS = 5;
+
+export class RecentlyAccessed extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false,
+    };
+  }
+
+  onButtonClick() {
+    this.setState({
+      isPopoverOpen: !this.state.isPopoverOpen,
+    });
+  }
+
+  closePopover() {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  }
+
+  renderDropdown = () => {
+    const dropdownLinks = [];
+    for (let i = NUM_LONG_LINKS; i < this.props.recentlyAccessed.length; i++) {
+      dropdownLinks.push(
+        (
+          <li
+            style={{ marginBottom: 8 }}
+            key={this.props.recentlyAccessed[i].link}
+          >
+            <EuiLink
+              className="dropwdownLink"
+              href={this.props.recentlyAccessed[i].link}
+            >
+              {this.props.recentlyAccessed[i].label}
+            </EuiLink>
+          </li>
+        )
+      );
+    }
+
+    const openPopoverComponent = (
+      <EuiLink onClick={this.onButtonClick.bind(this)}>
+        <EuiTextColor color="subdued">
+          {`+${dropdownLinks.length}`}
+        </EuiTextColor>
+        <EuiIcon
+          type="arrowDown"
+          color="subdued"
+        />
+      </EuiLink>
+    );
+
+    return (
+      <EuiFlexItem
+        key="dropdown"
+        grow={false}
+      >
+        <EuiPopover
+          id="popover"
+          ownFocus
+          button={openPopoverComponent}
+          isOpen={this.state.isPopoverOpen}
+          closePopover={this.closePopover.bind(this)}
+          anchorPosition="downRight"
+        >
+          <ul>
+            {dropdownLinks}
+          </ul>
+        </EuiPopover>
+      </EuiFlexItem>
+    );
+  }
+
+  renderLongLink = (link, label, includeSeperator = false) => {
+    let seperator;
+    if (includeSeperator) {
+      seperator = (
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type="dot"
+            color="subdued"
+          />
+        </EuiFlexItem>
+      );
+    }
+    // Want to avoid a bunch of white space around items with short labels (happens when min width is too large).
+    // Also want to avoid truncating really short names (happens when there is no min width)
+    // Dynamically setting the min width based on label lengh meets both of these goals.
+    const EM_RATIO = 0.6; // 'em' ratio that avoids too much horizontal white space and too much truncation
+    const minWidth = (label.length < 8 ? label.length : 8) * EM_RATIO;
+    const style = {
+      overflow: 'hidden',
+      minWidth: `${minWidth}em`
+    };
+    return (
+      <React.Fragment key={link}>
```

Comment:
Not sure how this happened, but I got this error due to conflicting keys:

![image](https://user-images.githubusercontent.com/1238659/36004900-d9ff35ca-0ce9-11e8-9823-c06324f15ffa.png)

Reply:
Good catch, looks like if you change the name of a saved object than the links are the same but the labels are different. I will make the react key be a combo of both to ensure no duplicate keys.